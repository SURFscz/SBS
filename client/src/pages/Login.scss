@import "../stylesheets/vars.scss";

.mod-login {

    background-color: $background;
    padding: 0;

    div.title {
        font-size: 22px;
        color: $dark-grey;
        max-width: $medium;
        margin: 20px auto;
        padding: 10px 0 5px 0;

        p {
            padding: 10px 0 10px 0;
        }

        a {
            display: inline-block;
            font-size: 18px;
            text-transform: uppercase;
            text-decoration: none;
            margin: 10px 0;
            color: $blue;

            &:hover {
                font-weight: bolder;
            }

            svg {
                margin-right: 10px;
            }
        }
    }

}
