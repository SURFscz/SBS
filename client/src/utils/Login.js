import {stopEvent} from "./Utils";
import {getParameterByName} from "./QueryParameters";
import {authorizationUrl, logoutUser} from "../api";

export function login(e, currentUrl = window.location.href) {
    stopEvent(e);
    const state = getParameterByName("state", window.location.search) || currentUrl;
    authorizationUrl(state).then(res => {
        window.location.href = res.authorization_endpoint;
    });
}

export function logout(e) {
    stopEvent(e);
    logoutUser().then(() => window.location.href = "/landing")
}

