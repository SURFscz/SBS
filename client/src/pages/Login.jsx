import React from "react";
import "./Login.scss";
import I18n from "i18n-js";
import {health} from "../api";
import {isEmpty, pseudoGuid, stopEvent} from "../utils/Utils";
import {getParameterByName} from "../utils/QueryParameters";
import {FontAwesomeIcon} from "@fortawesome/react-fontawesome";

class Login extends React.Component {

    componentDidMount = () => health();

    render() {
        const {user} = this.props;
        return (
            <div className="mod-login">
                <div className="title">
                    <p>{I18n.t("login.title")}</p>
                    {user.guest && <a href="/login" onClick={e => {
                    stopEvent(e);
                    const state = getParameterByName("state", window.location.search);
                    const guid = pseudoGuid();
                    const queryParameter = isEmpty(state) ? `?guid=${guid}` : `?guid=${guid}&state=${encodeURIComponent(state)}`;
                    window.location.href = `/login${queryParameter}`;
                }}><FontAwesomeIcon icon="arrow-right"/>{I18n.t("not_found.loginLink")}</a>}
                </div>
            </div>);
    };
}

export default Login;