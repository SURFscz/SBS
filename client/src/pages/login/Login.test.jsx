import React from "react";
import { render } from "@testing-library/react";
import Login from "./Login";

describe("Login", () => {
    it("matches snapshot with default props", () => {
        const { asFragment } = render(
            <Login />
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it("matches snapshot with custom title", () => {
        const { asFragment } = render(
            <Login title="Custom Login Title" />
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it("matches snapshot with custom description", () => {
        const { asFragment } = render(
            <Login description="Custom Login Description" />
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
