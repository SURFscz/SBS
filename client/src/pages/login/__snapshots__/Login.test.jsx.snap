// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login matches snapshot with custom description 1`] = `
<DocumentFragment>
  <div
    class="top-container"
  >
    <div
      class="mod-login-container"
    >
      <div
        class="mod-login"
      >
        <div
          class="header-left"
        >
          <h2
            class="header-title"
          >
            Easy and secure access to research applications for research collaborations
          </h2>
          <button
            class="sds--btn sds--btn--primary    "
            type="button"
          >
            <span
              class="textual"
            >
              Log in
            </span>
          </button>
          <p
            class="sup"
          >
            New to SURF Research Access Management? Log in and create a collaboration with your institution.
          </p>
        </div>
        <div
          class="header-right"
        >
          <img
            alt="logo"
            class=""
            src="happy.svg"
          />
        </div>
      </div>
    </div>
    <div
      class="mod-login-container bottom"
    >
      <div
        class="mod-login bottom"
      >
        <h1>
          How does it work?
        </h1>
        <div
          class="mod-login info reversed"
        >
          <div
            class="header-left info"
          >
            <div
              class="info-title"
            >
              <h2>
                Create
              </h2>
              <div
                class="admin-function-container"
              >
                <button
                  class="sds--chips main-400"
                >
                  <span
                    class="sds--chips--textual sds--text--body--small"
                  >
                    admin function
                  </span>
                </button>
              </div>
            </div>
            <p />
            <p>
              Administrators create a collaboration page for their community.
            </p>
            <p>
              They then select applications from the catalog to make them available to collaboration members.
            </p>
            <p />
          </div>
          <div
            class="header-right info"
          >
            <img
              alt="logo"
              class="reversed"
              src="sketch.svg"
            />
          </div>
        </div>
        <div
          class="mod-login info "
        >
          <div
            class="header-left info"
          >
            <div
              class="info-title"
            >
              <h2>
                Invite
              </h2>
              <div
                class="admin-function-container"
              >
                <button
                  class="sds--chips main-400"
                >
                  <span
                    class="sds--chips--textual sds--text--body--small"
                  >
                    admin function
                  </span>
                </button>
              </div>
            </div>
            <p />
            <p>
              Administrators invite fellow researchers to their collaboration page.
            </p>
            <p>
              They can also ask other administrators to help manage the collaboration.
            </p>
            <p />
          </div>
          <div
            class="header-right info"
          >
            <img
              alt="logo"
              class=""
              src="mail.svg"
            />
          </div>
        </div>
        <div
          class="mod-login info reversed"
        >
          <div
            class="header-left info"
          >
            <div
              class="info-title"
            >
              <h2>
                Join
              </h2>
            </div>
            <p />
            <p>
              Invited researchers join by logging in with their institutional account or eduID.nl.
            </p>
            <br />
            <p />
          </div>
          <div
            class="header-right info"
          >
            <img
              alt="logo"
              class="reversed"
              src="screen.svg"
            />
          </div>
        </div>
        <div
          class="mod-login info "
        >
          <div
            class="header-left info"
          >
            <div
              class="info-title"
            >
              <h2>
                Collaborate
              </h2>
            </div>
            <p />
            <p>
              The secured collaboration page shows its members and gives access to the research applications.
            </p>
            <p />
          </div>
          <div
            class="header-right info"
          >
            <img
              alt="logo"
              class=""
              src="collaborate.svg"
            />
          </div>
        </div>
        <div
          class="landing-footer"
        >
          <p />
          <p>
            SURF Research Access Management is a service for access management of Dutch led research collaborations.
          </p>
          <p>
            Do you want to know more? Please visit 
            <a
              href="https://surf.nl/en/sram"
            >
              https://surf.nl/en/sram
            </a>
            .
          </p>
          <p />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Login matches snapshot with custom title 1`] = `
<DocumentFragment>
  <div
    class="top-container"
  >
    <div
      class="mod-login-container"
    >
      <div
        class="mod-login"
      >
        <div
          class="header-left"
        >
          <h2
            class="header-title"
          >
            Easy and secure access to research applications for research collaborations
          </h2>
          <button
            class="sds--btn sds--btn--primary    "
            type="button"
          >
            <span
              class="textual"
            >
              Log in
            </span>
          </button>
          <p
            class="sup"
          >
            New to SURF Research Access Management? Log in and create a collaboration with your institution.
          </p>
        </div>
        <div
          class="header-right"
        >
          <img
            alt="logo"
            class=""
            src="happy.svg"
          />
        </div>
      </div>
    </div>
    <div
      class="mod-login-container bottom"
    >
      <div
        class="mod-login bottom"
      >
        <h1>
          How does it work?
        </h1>
        <div
          class="mod-login info reversed"
        >
          <div
            class="header-left info"
          >
            <div
              class="info-title"
            >
              <h2>
                Create
              </h2>
              <div
                class="admin-function-container"
              >
                <button
                  class="sds--chips main-400"
                >
                  <span
                    class="sds--chips--textual sds--text--body--small"
                  >
                    admin function
                  </span>
                </button>
              </div>
            </div>
            <p />
            <p>
              Administrators create a collaboration page for their community.
            </p>
            <p>
              They then select applications from the catalog to make them available to collaboration members.
            </p>
            <p />
          </div>
          <div
            class="header-right info"
          >
            <img
              alt="logo"
              class="reversed"
              src="sketch.svg"
            />
          </div>
        </div>
        <div
          class="mod-login info "
        >
          <div
            class="header-left info"
          >
            <div
              class="info-title"
            >
              <h2>
                Invite
              </h2>
              <div
                class="admin-function-container"
              >
                <button
                  class="sds--chips main-400"
                >
                  <span
                    class="sds--chips--textual sds--text--body--small"
                  >
                    admin function
                  </span>
                </button>
              </div>
            </div>
            <p />
            <p>
              Administrators invite fellow researchers to their collaboration page.
            </p>
            <p>
              They can also ask other administrators to help manage the collaboration.
            </p>
            <p />
          </div>
          <div
            class="header-right info"
          >
            <img
              alt="logo"
              class=""
              src="mail.svg"
            />
          </div>
        </div>
        <div
          class="mod-login info reversed"
        >
          <div
            class="header-left info"
          >
            <div
              class="info-title"
            >
              <h2>
                Join
              </h2>
            </div>
            <p />
            <p>
              Invited researchers join by logging in with their institutional account or eduID.nl.
            </p>
            <br />
            <p />
          </div>
          <div
            class="header-right info"
          >
            <img
              alt="logo"
              class="reversed"
              src="screen.svg"
            />
          </div>
        </div>
        <div
          class="mod-login info "
        >
          <div
            class="header-left info"
          >
            <div
              class="info-title"
            >
              <h2>
                Collaborate
              </h2>
            </div>
            <p />
            <p>
              The secured collaboration page shows its members and gives access to the research applications.
            </p>
            <p />
          </div>
          <div
            class="header-right info"
          >
            <img
              alt="logo"
              class=""
              src="collaborate.svg"
            />
          </div>
        </div>
        <div
          class="landing-footer"
        >
          <p />
          <p>
            SURF Research Access Management is a service for access management of Dutch led research collaborations.
          </p>
          <p>
            Do you want to know more? Please visit 
            <a
              href="https://surf.nl/en/sram"
            >
              https://surf.nl/en/sram
            </a>
            .
          </p>
          <p />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Login matches snapshot with default props 1`] = `
<DocumentFragment>
  <div
    class="top-container"
  >
    <div
      class="mod-login-container"
    >
      <div
        class="mod-login"
      >
        <div
          class="header-left"
        >
          <h2
            class="header-title"
          >
            Easy and secure access to research applications for research collaborations
          </h2>
          <button
            class="sds--btn sds--btn--primary    "
            type="button"
          >
            <span
              class="textual"
            >
              Log in
            </span>
          </button>
          <p
            class="sup"
          >
            New to SURF Research Access Management? Log in and create a collaboration with your institution.
          </p>
        </div>
        <div
          class="header-right"
        >
          <img
            alt="logo"
            class=""
            src="happy.svg"
          />
        </div>
      </div>
    </div>
    <div
      class="mod-login-container bottom"
    >
      <div
        class="mod-login bottom"
      >
        <h1>
          How does it work?
        </h1>
        <div
          class="mod-login info reversed"
        >
          <div
            class="header-left info"
          >
            <div
              class="info-title"
            >
              <h2>
                Create
              </h2>
              <div
                class="admin-function-container"
              >
                <button
                  class="sds--chips main-400"
                >
                  <span
                    class="sds--chips--textual sds--text--body--small"
                  >
                    admin function
                  </span>
                </button>
              </div>
            </div>
            <p />
            <p>
              Administrators create a collaboration page for their community.
            </p>
            <p>
              They then select applications from the catalog to make them available to collaboration members.
            </p>
            <p />
          </div>
          <div
            class="header-right info"
          >
            <img
              alt="logo"
              class="reversed"
              src="sketch.svg"
            />
          </div>
        </div>
        <div
          class="mod-login info "
        >
          <div
            class="header-left info"
          >
            <div
              class="info-title"
            >
              <h2>
                Invite
              </h2>
              <div
                class="admin-function-container"
              >
                <button
                  class="sds--chips main-400"
                >
                  <span
                    class="sds--chips--textual sds--text--body--small"
                  >
                    admin function
                  </span>
                </button>
              </div>
            </div>
            <p />
            <p>
              Administrators invite fellow researchers to their collaboration page.
            </p>
            <p>
              They can also ask other administrators to help manage the collaboration.
            </p>
            <p />
          </div>
          <div
            class="header-right info"
          >
            <img
              alt="logo"
              class=""
              src="mail.svg"
            />
          </div>
        </div>
        <div
          class="mod-login info reversed"
        >
          <div
            class="header-left info"
          >
            <div
              class="info-title"
            >
              <h2>
                Join
              </h2>
            </div>
            <p />
            <p>
              Invited researchers join by logging in with their institutional account or eduID.nl.
            </p>
            <br />
            <p />
          </div>
          <div
            class="header-right info"
          >
            <img
              alt="logo"
              class="reversed"
              src="screen.svg"
            />
          </div>
        </div>
        <div
          class="mod-login info "
        >
          <div
            class="header-left info"
          >
            <div
              class="info-title"
            >
              <h2>
                Collaborate
              </h2>
            </div>
            <p />
            <p>
              The secured collaboration page shows its members and gives access to the research applications.
            </p>
            <p />
          </div>
          <div
            class="header-right info"
          >
            <img
              alt="logo"
              class=""
              src="collaborate.svg"
            />
          </div>
        </div>
        <div
          class="landing-footer"
        >
          <p />
          <p>
            SURF Research Access Management is a service for access management of Dutch led research collaborations.
          </p>
          <p>
            Do you want to know more? Please visit 
            <a
              href="https://surf.nl/en/sram"
            >
              https://surf.nl/en/sram
            </a>
            .
          </p>
          <p />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
